ssz/codec.py

```diff
@@ -0,0 +1,37 @@
+from functools import (
+    lru_cache,
+)
+
+from eth_utils import (
+    is_bytes,
+)
+
+from ssz.exceptions import (
+    DecodingError,
+)
+from ssz.utils import (
+    infer_sedes,
+)
+
+
+@lru_cache(128)
+def encode(obj):
+    '''
+    Encode object in SSZ format
+    '''
+    serialized_obj = infer_sedes(obj).serialize(obj)
+    return serialized_obj
+
+
+@lru_cache(128)
+def decode(ssz, sedes=None):
+    '''
+    Decode a SSZ encoded object
+    '''
+    if not is_bytes(ssz):
+        raise DecodingError('Can only decode SSZ bytes, got type %s' % type(ssz).__name__, ssz)
+    if sedes:
+        obj = sedes.deserialize(ssz)
+        return obj
+
+    return ssz
```

Comment:
This looks weird and might be a bug, can you explain what's happening here? If we keep it like this, it would probably be a good idea to add a comment.

What do you think about making `sedes` required to eliminate this code path altogether. Would that have any negative side effects I'm missing?

Reply:
This codepath is because of the behaviour of `rlp` I was trying to mimic. Consider the following
```
In : rlp.decode(b'\x05')
Out : b'\x05'
```
Here in `rlp` they are not making `sedes` compulsory and are returning what was given if `sedes` was not mentioned. But I don't see any meaning in that, and maybe we could make `sedes` compulsory.

Reply:
I will be modifying the code, so that an `exception` is raised if `sedes` object is not specified while `deserialization`. Is that ok @jannikluhn